src/command_modules/azure-cli-monitor/azure/cli/command_modules/monitor/__init__.py

```diff
@@ -4,46 +4,76 @@
-        self.extra('namespace', options_list=['--{}-namespace'.format(dest)], arg_group=arg_group, help="Target resource provider namespace.")
-        self.extra('parent', options_list=['--{}-parent'.format(dest)], arg_group=arg_group, help="Target resource parent path, if applicable.")
-        self.extra('resource_type', options_list=['--{}-type'.format(dest)], arg_group=arg_group, help="Target resource type. Can also accept namespace/type format (Ex: 'Microsoft.Compute/virtualMachines)')")
+        self.argument(dest, options_list='--resource', arg_group=arg_group, required=required,
+                      validator=get_target_resource_validator(dest, required),
+                      help="Name or ID of the target resource.")
+        self.extra('namespace', options_list='--resource-namespace', arg_group=arg_group,
+                   help="Target resource provider namespace.")
+        self.extra('parent', options_list='--resource-parent', arg_group=arg_group,
+                   help="Target resource parent path, if applicable.")
+        self.extra('resource_type', options_list='--resource-type', arg_group=arg_group,
+                   help="Target resource type. Can also accept namespace/type format "
+                        "(Ex: 'Microsoft.Compute/virtualMachines)')")
+        self.extra('resource_group_name', options_list=('--resource-group', '-g'), arg_group='Target Resource')
 
 
 class MonitorCommandsLoader(AzCommandsLoader):
-
     def __init__(self, cli_ctx=None):
         super(MonitorCommandsLoader, self).__init__(cli_ctx=cli_ctx,
-                                                    min_profile='2017-03-10-profile',
                                                     argument_context_cls=MonitorArgumentContext)
 
+    def monitor_command_group(self, name, client_factory, sdk_module=None, sdk_class=None, custom_module=None,
+                              **kwargs):
+        command_type = CliCommandType(
+            operations_tmpl='azure.mgmt.monitor.operations.{}#{}.'.format(sdk_module, sdk_class) + '{}',
+            client_factory=client_factory) if (sdk_module and sdk_class) else None
+        custom_command_type = CliCommandType(
+            operations_tmpl='azure.cli.command_modules.monitor.operations.{}#'.format(custom_module) + '{}',
+            client_factory=client_factory) if custom_module else None
+
+        return self.command_group(name, command_type=command_type, custom_command_type=custom_command_type, **kwargs)
+
     def load_command_table(self, args):
         from azure.cli.command_modules.monitor.commands import load_command_table
         load_command_table(self, args)
+
         return self.command_table
 
     def load_arguments(self, command):
-        from azure.cli.command_modules.monitor._params import load_arguments
+        from azure.cli.command_modules.monitor.params import load_arguments
```

Comment:
Every single other module in the CLI has this file called _params.py.

Reply:
The inconsistence in naming the modules among a package is noticeable. Validators are public, commands are public, parameters are private.

I'd see a necessity to add underline before most of the module. And I don't see a necessity to force all module follows the same pattern, either. We should allow individual modules to varies on their own terms and it may introduce a better and more productive pattern.